packages/angular/ssr/public_api.ts

```diff
@@ -12,4 +12,6 @@ export {
   type CommonEngineOptions,
 } from './src/common-engine/common-engine';
 
+export { AngularAppEngine } from './src/app-engine';
```

Comment:
@AndrewKushnir, I recall you previously suggested that, instead of exporting `AngularAppEngine`, we might consider exporting a creator function like `createAngularAppEngine`. Do you think that approach would be necessary?

Reply:
As an option, we can also mimic the [`createPlatform`](https://angular.dev/api/core/createPlatform) and [`getPlatform`](https://angular.dev/api/core/getPlatform) APIs:

- the "create" function would only create an instance if no instance exists already and throw an error otherwise
- the "get" function would return the current instance of the engine or `null` if no engine is created yet

WDYT?

Reply:
I've updated the pattern to align with the conventions used for `ServerApp`. However, I'm considering whether the return type should be modified to an `AngularAppEngine` interface, ensuring that at least its public methods are included in the API golden.

Reply:
IE: https://github.com/angular/angular-cli/pull/28279/commits/588200adf9f22011f879c58180ab1ea76a4ec8a9